File: 2024/01/main.py
from sys import stdin

list_1 = []
list_2 = []

for index, line in enumerate(stdin.readlines()):
    values = line.strip().split()
    list_1.append(int(values[0]))
    list_2.append(int(values[1]))

total_distance = 0

for left, right in zip(sorted(list_1), sorted(list_2)):
    total_distance += abs(left - right)

similarity = 0

for left in list_1:
    similarity += list_2.count(left) * left

print(total_distance, similarity)

File: 2024/01/main.rs
use std::io::{self, Read};

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let mut input = String::new();
    io::stdin().read_to_string(&mut input)?;

    let mut left: Vec<i32> = vec![];
    let mut right: Vec<i32> = vec![];

    for line in input.lines() {
        let mut values = line.split_whitespace();

        left.push(values.next().unwrap().parse()?);
        right.push(values.next().unwrap().parse()?)
    }

    let mut total_distance = 0;

    left.sort();
    right.sort();

    for (left, right) in left.clone().into_iter().zip(&right) {
        total_distance += (left - right).abs();
    }

    let mut similarity = 0;

    for left in &left {
        similarity += right.iter().filter(|x| x == &left).count() as i32 * left;
    }

    println!("{} {}", total_distance, similarity);

    Ok(())
}
